### PR TITLE
fix cal

### PR DESCRIPTION
**Pull request by b97390022**
fix cal, pls help me check the new function : specialop for handling `**` and '^'

### CODE REVIEW

File: modules/calculator.py [hunk 1]
```diff
@@ -28,8 +40,13 @@ def cal(s, lop):
 def parse(s):
     if s == '':
         return 'Empty string!'
+    if s.count('(') != s.count(')'):
```
Comment (b97390022) [inline modules/calculator.py:43]: Here is to prevent unpair of '(' and ')'

File: modules/calculator.py [hunk 2]
```diff
@@ -70,4 +87,32 @@ def handle(s):
                 elif i == 0:
                     s = f'-{s[i:p]}/{s[p+2:]}'
     return s
+def specialop(s):
```
Comment (b97390022) [inline modules/calculator.py:90]: new function to handle situation of '^' and '**' (positive power and negative power)

Comment (MBB200291) [reply modules/calculator.py:90]: image
image

Comment (MBB200291) [reply modules/calculator.py:90]: 有bug耶

**Review by linkuanhang — DISMISSED** [review]
ValueError for "2**(2-3)"

**Review by linkuanhang — COMMENTED** [review]
There is a wrong result for "(2-3)**2/3". Expect with 1/3, but get -1/3.

**Comment by MBB200291** [comment]
image
冠翰說的問題還在耶

**Comment by linkuanhang** [comment]
Make a new one for calculator.

**Pull request closed without merging**